File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
      int open=0,max=0;
      for(char ch:s.toCharArray()){
          if(ch=='('){
              open++;
          }
          else if(ch==')'){
              open--;
          }
          max=Math.max(max,open);
      }  
      return max;
    }
}

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        open=0
        max_val=float('-inf')
        for c in s:
            if c=='(':
                open+=1
            elif c==')':
                open-=1
            max_val=max(max_val,open)
        return max_val